runtime-testsuite/test/org/antlr/v4/test/runtime/typescript/TscRunner.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2012-2017 The ANTLR Project. All rights reserved.
+ * Use of this file is governed by the BSD 3-clause license that
+ * can be found in the LICENSE.txt file in the project root.
+ */
+package org.antlr.v4.test.runtime.typescript;
+
+import org.antlr.v4.test.runtime.*;
+import org.antlr.v4.test.runtime.states.CompiledState;
+import org.antlr.v4.test.runtime.states.GeneratedState;
+import org.stringtemplate.v4.ST;
+
+import java.io.File;
+
+import static org.antlr.v4.test.runtime.FileUtils.writeFile;
+
+public class TscRunner extends RuntimeRunner {
+
+	/* TypeScript runtime is the same as JavaScript runtime */
+	private final static String NORMALIZED_JAVASCRIPT_RUNTIME_PATH = getRuntimePath("JavaScript").replace('\\', '/');
+
+	@Override
+	public String getLanguage() {
+		return "TypeScript";
+	}
+
+	@Override
+	protected void initRuntime() throws Exception {
+		npmLinkRuntime();
```

Comment:
I guess we have different views on code quality. It is much better to put the code in a function because it self-documents the code, and when one needs to add code in init they can add a call to another method. <init> is self-explanatory of _when_ it is called, but not on _what_ it does.

Reply:
From my point of view, it's obvious what the following code does, it's self-explanatory:

```java
Processor.run(new String[] {"npm", "--silent", "link"}, NORMALIZED_JAVASCRIPT_RUNTIME_PATH);
```

There is no new info from method name `npmLinkRuntime`.

> when one needs to add code in init they can add a call to another method.

At present time, there is no other code inside the method, in future it can be another situation, but it dosn't matter for now.